src/pages/item/ItemInfo.jsx

```diff
@@ -1,39 +1,121 @@
 import { useParams } from "react-router";
+import { data } from "../../data/data";
 import { Link } from "react-router-dom";
+import { RiHeartAddLine } from "react-icons/ri";
+import { useFavorites } from "../../stogre/addToLikes";
 import "./ItemInfo.css";
-import { AiOutlineHeart } from "react-icons/ai";
+import { useColorChanging } from "../../stogre/colorChangeing";
+
+// ***** component start ***** //
 
 const ItemInfo = () => {
+  const colors = useColorChanging();
+
+  const favoritesContext = useFavorites();
+  const toggleFavoriteStatusHandler = (id) => {
+    const changedId = parseInt(id);
+    const isFavorite = favoritesContext.itemIsFavorite(changedId);
+    const userFavorite = data.filter((product) => product.id === changedId);
+    if (isFavorite) {
+      favoritesContext.removeFavorite(changedId);
+    } else {
+      favoritesContext.addFavorite(userFavorite);
+    }
+    console.log(favoritesContext.favorites);
+  };
   const [product, setProduct] = useState("default droduct");
   const [count, setCount] = useState(0);
   const { id } = useParams();
   useEffect(() => {
     const newProduct = data.find((product) => product.id === parseInt(id));
     setProduct(newProduct);
   }, [id]);
-  const { title, img, price, details } = product;
+
+  const { title, img1, img2, img3, price, details } = product;
+
+  const imageHandler = () => {
+    if (colors.isBlack) {
+      return img1;
+    }
+    if (colors.isGold) {
+      return img2;
+    }
+    if (colors.isBlue) {
+      return img3;
+    }
+  };
+
+  const params = {
+    slidesPerView: "auto",
+    pagination: {
+      el: ".swiper-pagination",
+    },
+  };
   return (
     <main className="item-page">
+      <header className="header-of-item">
+        <Link key="1" to="/listofproducts">
+          <FiChevronLeft />
+        </Link>
+        <Link key="2" to="/card">
+          <BiShoppingBag />
+        </Link>
+      </header>
       <div className="item-img-container">
-        <header className="header-of-item">
-          <Link key="1" to="/listofproducts">
-            <FiChevronLeft />
+        <Swiper {...params}>
+          <div className="item-single-pic">
+            <img
+              src={imageHandler()}
+              alt={title}
+              className="headphone-pic-item"
+              id="1"
+            />
+          </div>
+          <div className="item-single-pic">
+            <img
+              src={imageHandler()}
+              alt={title}
+              className="headphone-pic-item"
+              id="2"
+            />
+          </div>
+          <div className="item-single-pic">
+            <img
+              src={imageHandler()}
+              alt={title}
+              className="headphone-pic-item"
+              id="3"
+            />
+          </div>
+        </Swiper>
+        <div className="item-change-color">
+          <Link className="color-link1" onClick={colors.colorIsBlack}>
```

Comment:
Why you are using `Link` component, this will not redirect us to other pages. simply you can use button and don't need that empty `<p />` elements